packages/contracts/src/contracts/current/protocol/Exchange/MixinSettlement.sol

```diff
@@ -96,4 +100,91 @@ contract MixinSettlement is
         );
         return (makerAssetFilledAmount, makerFeePaid, takerFeePaid);
     }
+
+    /// @dev Settles matched order by transferring appropriate funds between order makers, taker, and fee recipient.
+    /// @param leftOrder First matched order.
+    /// @param rightOrder Second matched order.
+    /// @param matchedFillResults Struct holding amounts to transfer between makers, taker, and fee recipients.
+    /// @param takerAddress Address that matched the orders. The taker receives the spread between orders as profit.
+    function settleMatchedOrders(
+        LibOrder.Order memory leftOrder,
+        LibOrder.Order memory rightOrder,
+        MatchedFillResults memory matchedFillResults,
+        address takerAddress)
+        internal
+    {
+        // Optimized for:
+        // * leftOrder.feeRecipient =?= rightOrder.feeRecipient
+
+        // Not optimized for:
+        // * {left, right}.{MakerAsset, TakerAsset} == ZRX
+        // * {left, right}.maker, takerAddress == {left, right}.feeRecipient
+
+        // leftOrder.MakerAsset == rightOrder.TakerAsset
+        // Taker should be left with a positive balance (the spread)
+        dispatchTransferFrom(
```

Comment:
Currently, transfers look like:
leftMaker => taker
taker => rightMaker
rightMaker => leftMaker
The spread is always coming from the leftMaker.

However, we never want the taker to actually take custody of the tokens (other than the spread). So what would be more desirable would be:
leftMaker => rightMaker
rightMaker => leftMaker
leftMaker => taker